### ISSUE TITLE
Investigate Django Usability

### ISSUE THREAD

**Issue by dishad** [dishad/deanslist#1]
Looks like using Django may be in our best interest as it will allow our development to be pretty fast and has easy integration for databases. Drawback is that we are going to need to learn how to use Python in this environment, but I think that since the TEP National Site that Brad is working on uses this framework whatever knowledge we get from working on this will be applicable to the National site as well.

Here is a link to what they offer:
https://www.djangoproject.com/start/overview/

**Comment by aqandrew** [comment]
And vice versa! I started reading up on Django this morning after building the TEP National site on my machine and it's sick. The fact that you can set up a database with like 10 Python lines blows my mind.

This overview of a sample project structure makes it out to be like MVC but with a lot of useful fluff:
https://docs.djangoproject.com/en/1.9/intro/overview/

**Comment by aqandrew** [comment]
Plus we can reference the National site codebase for examples of Django stuff; Shank and Brad are open to questions.

**Comment by aqandrew** [comment]
https://docs.djangoproject.com/en/1.10/ref/templates/language/#template-inheritance

### tl;dr
> One common way of using inheritance is the following three-level approach:
>
> - Create a base.html template that holds the main look-and-feel of your site.
> - Create a base_SECTIONNAME.html template for each “section” of your site. For example, base_news.html, base_sports.html. These templates all extend base.html and include section-specific styles/design.
> - Create individual templates for each type of page, such as a news article or blog entry. These templates extend the appropriate section template.
> This approach maximizes code reuse and makes it easy to add items to shared content areas, such as section-wide navigation.

This particular section of Django's documentation gives a good overview of how its template system works--an understanding of this will make it easier to map out the site.

**Comment by aqandrew** [comment]
I used this guide for basic setup of the app.

http://opentechschool.github.io/django-101/en/first_steps/index.html
